### DISCUSSION TITLE
Set Tiff Tags

### DISCUSSION THREAD

**Discussion by yaremam** [dlemstra/Magick.NET#868] [Help]
I am using `image.SetAttribute("tiff:software", "mysoftware")` but the tag is not saved to a tiff image when I write it. With imageMagick I can use ` -set tiff:software "mysoftware"` . Is there a way to do this in Magick.NET?

This didn't work either:

                 var profile = image.GetExifProfile();

                if (profile == null)
                {
                    profile = new ExifProfile();
                }

                profile.SetValue(ExifTag.Software, "MySoftware");

                image.SetProfile(profile);`

**Comment by dlemstra** [comment]
There seems to be a typo in your `SetAttribute` call? It should be `"tiff:software"` instead.

**Reply by yaremam** [reply]
Just a typo,  tiff:software does not work.  `image.GetAttribute` just before `Write `returns correct value, it seems Write doesn't actually write it.

**Reply by dlemstra** [reply]
You will need to use `image.SetArtifact("tiff:software", "MySoftware");`
